### ISSUE TITLE
Railties tests using `generator.invoke_all` install Rails to gem path

### ISSUE THREAD

**Issue by yahonda** [rails/rails#31557]
While investigating #31531 There are 4 tests which install railties (Rails) to the gem path,
which trigger #31531 not reproduced.

#31531 itself has been addressed, however, these 4 tests should not install Rails to the gem path.

### Steps to reproduce

```ruby
$ rbenv install 2.4.3 ; rbenv global 2.4.3 (or whatever way installing Ruby)
$ gem install bunlder
$ gem env
... snip 
  - GEM PATHS:
     - /home/yahonda/.rbenv/versions/2.4.3/lib/ruby/gems/2.4.0
     - /home/yahonda/.gem/ruby/2.4.0
... snip 
$ git clone https://github.com/rails/rails.git
$ cd rails/railties
$ bundle install
```

* Execute one of these 4 tests. Once any of them executed, ruby environment `gem path` needs clean up to remove Rails installed to the gem path.

```ruby
$ bundle exec ruby -Itest test/generators/app_generator_test.rb -n test_active_storage_install
$ bundle exec ruby -Itest test/generators/app_generator_test.rb -n test_webpack_option_with_js_framework
$ bundle exec ruby -Itest test/generators/app_generator_test.rb -n test_template_is_executed_when_supplied_an_https_path
$ bundle exec ruby -Itest test/generators/app_generator_test.rb -n test_webpack_option
```

### Expected behavior
All of them should use Rails environment tested.

### Actual behavior
These tests successfully finished, however they are using Rails installed to the gem path.

### System configuration
**Rails version**:master branch

**Ruby version**:ruby 2.4.3p205 (2017-12-14 revision 61247) [x86_64-linux]

**Comment by y-yagi** [comment]
Fixed by #31564

**Comment by yahonda** [comment]
Thanks. I have validated #31564 has been addressed.
